File: DotNet/Messaging/Attachments/StringAttachment.cs

using System.Net.Http;

namespace Shastra.Hydra.Messaging.Attachments
{
    public class StringAttachment : Attachment
    {
        private const string TextContentType = "text/plain";

        public string Data { get; private set; }

        public StringAttachment(string name, string data) : base(name, TextContentType)
        {
            Data = data;
        }

        #region Overrides of Attachment

        public override int DataLength() { return Data.Length; }

        public override HttpContent ToHttpContent() { return new StringContent(Data); }

        #endregion
    }
}

File: Java/hydra/src/main/java/uk/co/shastra/hydra/messaging/attachments/StringAttachment.java
package uk.co.shastra.hydra.messaging.attachments;

import org.apache.http.entity.ContentType;
import org.apache.http.entity.mime.content.ContentBody;
import org.apache.http.entity.mime.content.StringBody;

public class StringAttachment extends Attachment {

    private static final String TextContentType = "text/plain";
    private String data;

	public String getData() { return data; }
	
	public StringAttachment(String name, String data) {
		super(name, TextContentType);
		this.data = data;
	}

	@Override
	public int dataLength() { return data.length(); }

	@Override
	public ContentBody toContentBody() { return new StringBody(data, ContentType.TEXT_PLAIN); }

}
